frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/SelectionResult.java

```diff
@@ -18,30 +18,56 @@
 package com.microsoft.frameworklauncher.applicationmaster;
 
 import com.microsoft.frameworklauncher.common.exts.CommonExts;
+import com.microsoft.frameworklauncher.common.model.Range;
+import com.microsoft.frameworklauncher.common.utils.ValueRangeUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
 
 public class SelectionResult {
-  private String nodeHost;
-  private Long gpuAttribute;
 
-  public String getNodeHost() {
-    return nodeHost;
+  private Map<String, Long> selectedNodes = new HashMap<String, Long>();
```

Comment:
>selectedNodes [](start = 28, length = 13)

It would better to be:
// hostName -> GpuAttribute
  private Map<String, Long> selectedGpuAttributes= new HashMap<String, Long>();